utils/polus-bfio-util/bfio/bfio.py

```diff
@@ -375,171 +621,361 @@ def _init_writer(self):
+
+        Args:
+            Y (int, optional): height of image in pixels. Defaults to None.
+
+        Returns:
+            int: height of image in pixels
+        """
         if Y:
             assert not self.__writer, "The image has started to be written. To modify the xml again, reinitialize."
-            assert Y>=1
+            assert Y >= 1
             self._metadata.image(0).Pixels.SizeY = Y
             self._xyzct['Y'] = Y
-        else:
-            return self._xyzct['Y']
-    def num_z(self,Z=None):
+        
+        return self._xyzct['Y']
+
+    def num_z(self, Z=None):
+        """num_z Get/Set the number of pixels in the z-dimension
+
+        If Z is defined, it sets the number of pixels in the z-dimension
+        (i.e. the image depth). If Z is None, return the image depth.
+
+        Args:
+            Z (int, optional): Depth of image in pixels. Defaults to None.
+
+        Returns:
+            int: depth of image in pixels
+        """
         if Z:
             assert not self.__writer, "The image has started to be written. To modify the xml again, reinitialize."
-            assert Z>=1
+            assert Z >= 1
             self._metadata.image(0).Pixels.SizeZ = Z
             self._xyzct['Z'] = Z
-        else:
-            return self._xyzct['Z']
-    def num_c(self,C=None):
+        
+        return self._xyzct['Z']
+
+    def num_c(self, C=None):
+        """num_c Get/Set the number of channels
+
+        If C is defined, set the number of image channels. If C is
+        None, return the number of channels.
+
+        Args:
+            C (int, optional): Number of image channels. Defaults to None.
+
+        Returns:
+            int: Number of image channels
+        """
         if C:
             assert not self.__writer, "The image has started to be written. To modify the xml again, reinitialize."
-            assert C>=1
+            assert C >= 1
             self._metadata.image(0).Pixels.SizeC = C
             self._xyzct['C'] = C
-        else:
-            return self._xyzct['C']
-    def num_t(self,T=None):
+        
+        return self._xyzct['C']
+
+    def num_t(self, T=None):
+        """num_t Get/Set the number of timepoints
+
+        If T is defined, set the number of image timepoints. If C is
+        None, return the number of channels.
+
+        Args:
+            C (int, optional): Number of image channels. Defaults to None.
+
+        Returns:
+            int: Number of image channels
+        """
         if T:
             assert not self.__writer, "The image has started to be written. To modify the xml again, reinitialize."
-            assert T>=1
+            assert T >= 1
             self._metadata.image(0).Pixels.SizeT = T
             self._xyzct['T'] = T
-        else:
-            return self._xyzct['T']
         
+        return self._xyzct['T']
+
     def physical_size_x(self, psize=None, units=None):
```

Comment:
Actually since you don't use series anywhere else but in the bioreader class, should we include it at all? I am thinking it has value in tomographic data (and in absorbance of course!) but since it looks like it is out of scope for bioformats, do we want to include it in any of the bfio tool?

Reply:
I can actually just get rid of it. Series has more to do with embedding multiple images within a single TIFF from the Bioformats side of things.